k-distribution/include/builtin/domains.k

```diff
@@ -177,11 +178,20 @@ module MAP-SYMBOLIC [symbolic]
   // potential infinite loop
   // rule (MAP:Map [ K1 <- V1 ]) [ K2 <- V2 ] => MAP [ K2 <- V2 ] [ K1 <- V1 ] requires K1 =/=K K2
 
-  rule K1 in_keys(M K2 |-> _) => true          requires K1  ==K K2 orBool K1 in_keys(M)
-  rule K1 in_keys(M K2 |-> _) => K1 in_keys(M) requires K1 =/=K K2
+  rule K in_keys(M K |-> _) => true
+  rule K in_keys(_) => false [owise]
+
+  rule K1 in_keys(M K2 |-> _) => true          requires K1  ==K K2 orBool K1 in_keys(M) [simplification]
+  rule K1 in_keys(M K2 |-> _) => K1 in_keys(M) requires K1 =/=K K2 [simplification]
```

Comment:
Well, I figured that since the Java engine still needs to work, we must keep these around.
One way to exclude them from the Haskell backend all-together would be to put them into a module which has attributes `symbolic` and `kast`, which would make it acceptable to the Java backend but excluded from the Haskell backend.  I'll try and do that.

Reply:
I've completely separated the existing symbolic part from the Haskell part. We can decide later whether we want to share some of the rules or not.